pkg/_pub_shared/lib/utils/http.dart

```diff
@@ -47,23 +47,41 @@ Future<K> httpGetWithRetry<K>(
   Uri uri, {
   required FutureOr<K> Function(http.Response response) responseFn,
   int maxAttempts = 3,
+
+  /// The HTTP client to use.
+  ///
+  /// Note: when the client is specified, the inner loop will not create a new client object on retries.
```

Comment:
I would formulate this the other way around
```suggestion
  /// Note: when the client is not specified, the inner loop will create a new [Client] object on each retry attempt.
```